src/harness/testcases/WINNF_FT_S_GRA_testcase.py

```diff
@@ -133,4 +134,87 @@ def test_WINNF_FT_S_GRA_7(self):
+  def test_WINNF_FT_S_GRA_10(self):
+    """First request for 2 devices granted as PAL and GAA channel,
+    send next request for PAL and GAA channel for the same frequency range.
 
+    Response Code '0' for first request and '401' for next request.
+    """
+    # Load two devices.
+    device_a = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    device_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+
+    # Load data
+    pal_record = json.load(
+        open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+    pal_low_frequency = 3550000000
+    pal_high_frequency = 3560000000
+    ppa_record = json.load(
+        open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+    ppa_record, pal_record = makePpaAndPalRecordsConsistent(ppa_record,
+                                                            [pal_record],
+                                                            pal_low_frequency,
+                                                            pal_high_frequency,
+                                                            device_a['userId'])
+
+    # Insert device_a into the PPA zone.
+    device_a['installationParam']['latitude'], device_a['installationParam'][
+        'longitude'] = getRandomLatLongInPolygon(ppa_record)
+    # Register 2 devices
+    cbsd_ids = self.assertRegistered([device_a, device_c])
+
+    # Update PPA Record with CBSD ID and Inject data.
+    ppa_record['ppaInfo']['cbsdReferenceId'] = [cbsd_ids[0]]
+    self._sas_admin.InjectPalDatabaseRecord(pal_record[0])
+    zone_id = self._sas_admin.InjectZoneData({'record': ppa_record})
+    self.assertTrue(zone_id)
+
+    # Create grant request
+    grant_0 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_0['cbsdId'] = cbsd_ids[0]
+    grant_0['operationParam']['operationFrequencyRange'] = {
+        'lowFrequency': pal_low_frequency,
+        'highFrequency': pal_high_frequency
+    }
+    grant_1 = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    grant_1['cbsdId'] = cbsd_ids[1]
+    grant_1['operationParam']['operationFrequencyRange'] = {
+        'lowFrequency': 3560000000,
+        'highFrequency': 3570000000
+    }
+    request = {'grantRequest': [grant_0, grant_1]}
+
+    # Send grant request first time
+    response = self._sas.Grant(request)['grantResponse']
+    # Check grant response, must be response code 0.
+    self.assertEqual(len(response), 2)
+    for response_num in [0, 1]:
+      self.assertEqual(response[response_num]['cbsdId'], cbsd_ids[response_num])
+      self.assertTrue('grantId' in response[response_num])
+      self.assertTrue('grantExpireTime' in response[response_num])
+      datetime.strptime(response[response_num]['grantExpireTime'],
+                        '%Y-%m-%dT%H:%M:%SZ')
+      self.assertTrue('heartbeatInterval' in response[response_num])
```

Comment:
The TC says "The heartbeatInterval parameter set to valid value with correct format. "   but it doesn't specify what a valid value, do they mean >0?  should we just check 'heartbeatInterval'>0?  since we are not transmitting yet I don't believe a maximum value is enforced at this point.

Reply:
Yes, Thanks for spotting it. Added 'heartbeatInterval' format checks. Could you please take a look.